cdap-docs/reference-manual/source/http-restful-api/metadata.rst

```diff
@@ -573,11 +562,123 @@ Here is an example, pretty-printed::
    * - ``404 NOT FOUND``
      - No entities matching the specified query were found
 
+.. _http-restful-api-metadata-run:
+
+Retrieving Metadata for a Program run
+-------------------------------------
+At every run of a program, the metadata associated with the program, the application it is part of, and any datasets
+and streams used by the program run is recorded. To retrieve the metadata for a program run, submit an HTTP GET request::
+
+  GET <base-url>/namespaces/<namespace>/apps/<app-id>/<program-type>/<program-id>/runs/<run-id>/metadata/tags
+
+with the metadata returned as a JSON string in the return body::
+
+  [
+      {
+          "entityId": {
+              "id": {
+                  "namespace": {
+                      "id": "default"
+                  },
+                  "streamName": "purchaseStream"
+              },
+              "type": "stream"
+          },
+          "properties": {},
+          "scope": "USER",
+          "tags": []
+      },
+      {
+          "entityId": {
+              "id": {
+                  "application": {
+                      "applicationId": "PurchaseHistory",
+                      "namespace": {
+                          "id": "default"
+                      }
+                  },
+                  "id": "PurchaseFlow",
+                  "type": "Flow"
+              },
+              "type": "program"
+          },
+          "properties": {},
+          "scope": "USER",
+          "tags": [
+              "flow-tag1"
+          ]
+      },
+      {
+          "entityId": {
+              "id": {
+                  "instanceId": "purchases",
+                  "namespace": {
+                      "id": "default"
+                  }
+              },
+              "type": "datasetinstance"
+          },
+          "properties": {},
+          "scope": "USER",
+          "tags": []
+      },
+      {
+          "entityId": {
+              "id": {
+                  "applicationId": "PurchaseHistory",
+                  "namespace": {
+                      "id": "default"
+                  }
+              },
+              "type": "application"
+          },
+          "properties": {},
+          "scope": "USER",
+          "tags": [
+              "app-tag1"
+          ]
+      }
+  ]
+
+.. list-table::
+:widths: 20 80
+   :header-rows: 1
+
+     * - Parameter
```

Comment:
some of these parameters do not appear in the URL above. Perhaps you should say, it has one of the following forms: 
<url pattern for programs>
<url pattern for datasets>
<url pattern for streams>